### ISSUE TITLE
Webpack Integration - 'Cannot resolve module 'child_process'

### ISSUE THREAD

**Issue by OAGr** [cytoscape/cytoscape.js#998]
I'm trying to use cytoscape.js with webpack and am having trouble including it.  I've tried both:
```
var cytoscape = require('cytoscape');
```
and 
```
import cytoscape from 'cytoscape'
```

In both cases, running 'webpack' creates the errors: 
```
WARNING in ./~/cytoscape/dist/cytoscape.js
Critical dependencies:
563:63-70 require function is used in a way in which dependencies cannot be statically extracted
577:59-66 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/cytoscape/dist/cytoscape.js 563:63-70 577:59-66

WARNING in ./~/cytoscape/dist/cytoscape.js
Critical dependencies:
564:16-31 the request of a dependency is an expression
578:10-583:12 the request of a dependency is an expression
 @ ./~/cytoscape/dist/cytoscape.js 564:16-31 578:10-583:12

WARNING in ./~/cytoscape/dist/cytoscape.min.js
Critical dependencies:
18:1282-1289 require function is used in a way in which dependencies cannot be statically extracted
18:1157-1164 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/cytoscape/dist/cytoscape.min.js 18:1282-1289 18:1157-1164

WARNING in ./~/cytoscape/dist/cytoscape.min.js
Critical dependencies:
18:1169-1179 the request of a dependency is an expression
18:1291-1344 the request of a dependency is an expression
 @ ./~/cytoscape/dist/cytoscape.min.js 18:1169-1179 18:1291-1344

ERROR in ./~/cytoscape/dist/cytoscape.js
Module not found: Error: Cannot resolve module 'child_process' in /home/ozzie/personal/test/node_modules/cytoscape/dist
 @ ./~/cytoscape/dist/cytoscape.js 7170:30-54

ERROR in ./~/cytoscape/dist/cytoscape.min.js
Module not found: Error: Cannot resolve module 'child_process' in /home/ozzie/personal/test/node_modules/cytoscape/dist
 @ ./~/cytoscape/dist/cytoscape.min.js 21:10865-10889
```

Very similar issues have been documented in other npm packages:
https://github.com/ParsePlatform/ParseReact/issues/11
https://github.com/webpack/webpack-dev-server/issues/66

**Comment by OAGr** [comment]
I was able to get around this by doing,
```
module.exports.node = {
  child_process: 'empty'
}
```
Of course, this doesn't seem like the best fix.

**Comment by maxkfranz** [comment]
Cytoscape.js uses `child_process` only in Node.js/io.js environments, and this is detected dynamically.  If your tool is doing static `require()` analysis, you'll have to config your tool to ignore things like that -- as static `require()` analysis is insufficient for determining when/if actual `require()` calls are made.

Why isn't the tool looking at something explicit like `package.json` which lists no package dependencies?

**Comment by maxkfranz** [comment]
@OAGr 

Was able to find a workaround : https://gitter.im/webpack/webpack/archives/2014/08/26

Will be built into `package.json` for 2.4.5

**Comment by swillard-r7** [comment]
Any ETA on a 2.4.5 release? Ran into this issue, too.

**Comment by maxkfranz** [comment]
@swillard-r7 The current release is 2.6.5.

**Comment by swillard-r7** [comment]
Whoops -- yeah sorry, was depending on an old version, thanks!

**Comment by panuganti** [comment]
@OAGr  where did you specify 
module.exports.node = {
  child_process: 'empty'
}

**Comment by jogshraddha** [comment]
> _Module not found: Error: Cannot resolve module 'child_process' in /project/node_modules/serialport/lib/bindings

I am facing above error when I build my react application. My application is created using create-react-app and electronjs. I googled for this issue and the most suggested solutions are related to webpack config which I can not directly modify in create-react-app. I tried those solutions like adding 'target: node' and external config etc by ejecting create react app but that also did not work. Before adding serialport package it was working properly now it looks up for the child_process package into serialport libs. Can someone help me out in this?
